src/libsync/propagatedownload.cpp

```diff
@@ -357,12 +357,11 @@ void PropagateDownloadFile::start()
         }
     }
 
-    // If we have a conflict where size and mtime are identical,
+    // If we have a conflict where size of the file is unchanged,
     // compare the remote checksum to the local one.
     // Maybe it's not a real conflict and no download is necessary!
     if (_item->_instruction == CSYNC_INSTRUCTION_CONFLICT
         && _item->_size == _item->_previousSize
-        && _item->_modtime == _item->_previousModtime
```

Comment:
I'd change that to

`_item->_modtime == _item->_previousModtime  || isChecksumSufficientlyGood`